### PR TITLE
Extract transforms for lerna

### PR DESCRIPTION
**Pull request by rgbkrk**
Extract our transforms from media type payloads ➡️  React Elements.

### CODE REVIEW

**Comment by ivanov** [comment]
rebased, for starters.

File: packages/transforms-full/package.json [hunk 1]
```diff
@@ -26,5 +26,12 @@
   },
   "devDependencies": {
     "react": "^15.4.2"
+  },
+  "dependencies": {
+    "@nteract/transform-geojson": "1.0.0",
+    "@nteract/transform-model-debug": "1.0.0",
+    "@nteract/transform-plotly": "1.0.0",
+    "@nteract/transform-vega": "1.0.0",
+    "@nteract/transforms":  "1.0.0"
```
Comment (rgbkrk) [inline packages/transforms-full/package.json:35]: ❤️  this is definitely moving along

**Comment by ivanov** [comment]
Tests pass for me locally, let's see what our CI pals have to say about it.

**Comment by ivanov** [comment]
Oh, I think it's because I ran lerna locally, which CI doesn't yet know about. Let me add that.

**Comment by ivanov** [comment]
I think the flow errors we're seeing are similar to some of the issues reported on this thread: https://github.com/facebook/immutable-js/issues/203 but I have not followed it all to figure out how to get around it.

**Comment by ivanov** [comment]
I don't really know how flow works, but might it have something to do with the fact that we have 3 copies / version of immutable.js? 

```
$ find . -name immutable.js.flow
./node_modules/immutable/dist/immutable.js.flow
./packages/commutable/node_modules/immutable/dist/immutable.js.flow
./packages/transforms/node_modules/immutable/dist/immutable.js.flow
```

so the lerna-ed packages reference their own, but the main app references its own, and flow doesn't understand that they're the same?

**Comment by rgbkrk** [comment]
I'm looking at the flow types for Immutable and the `getIn` definition is _very_ wrong:

```
getIn<T>(searchKeyPath: ESIterable<any>, notSetValue: T): T;
getIn<T>(searchKeyPath: ESIterable<any>): T;
```

A deeply nested value may not actually be `T`.

@peggyrayzis what's the best way for us to ignore bad parts of definitions that come as part of the library (not in our `flow-typed` directory)?

**Comment by rgbkrk** [comment]
Then again, this was passing on master.

Have you run into anything like this @thejameskyle or @hzoo? We're trying to adapt to using lerna (amidst shipping an Electron app), running into trouble with the flow definitions for Immutable.

**Comment by ivanov** [comment]
For what it's worth, I get the same flow error on master. :\

**Comment by ivanov** [comment]
Ok, so here's how the errors can be eliminated: `lerna bootstrap --hoist immutable`... - This will not put `immutable` in the packages/*/node_modules, but only in the top level node_modules. 

except I think there's a bug in lerna, because running that with the `--hoist immutable` flag it just hangs there... 

symlinking the root `node_modules/immutable` to the `packages/commutable/node_modules/immutable` also fixes the flow stuff. As does removing the latter after a `lerna bootstrap` by hand.

**Comment by peggyrayzis** [comment]
>@peggyrayzis what's the best way for us to ignore bad parts of definitions that come as part of the library (not in our flow-typed directory)?

It looks like Paul already fixed the issue, but I would put a line in the [ignore] section of `.flowconfig` like `.*/node_modules/immutable/dist/.*`. This is an OCaml regex, so it can can pick up on any subdirectories, like the directories in `packages/`, that also contain this file pattern.

**Comment by rgbkrk** [comment]
> For what it's worth, I get the same flow error on master. :\

Ah, that must be because you and I have both done a `lerna bootstrap`, which means that it's resolving the Immutable in its `node_modules`.

**Comment by rgbkrk** [comment]
> It looks like Paul already fixed the issue, but I would put a line in the [ignore] section of .flowconfig like .*/node_modules/immutable/dist/.*. This is an OCaml regex, so it can can pick up on any subdirectories, like the directories in packages/, that also contain this file pattern.

Tried it, didn't work. 😭

**Comment by rgbkrk** [comment]
I wonder if this is because we listed it as a dev dependency:

https://github.com/lerna/lerna/issues/505

We could switch back to a normal dependency, the reason I didn't want to go that route was so that people could rely on it as a peer dependency.

**Comment by rgbkrk** [comment]
I'm trying some things and will post a new commit when I think it's in a friendly state.

**Comment by rgbkrk** [comment]
It passes! 🔨

File: package.json [hunk 2]
```diff
@@ -31,7 +31,7 @@
     "lint": "eslint src/ test/",
     "lint:fix": "eslint . --fix",
     "prebuild": "rimraf lib",
-    "lerna": "lerna bootstrap --hoist immutable",
+    "lerna": "lerna bootstrap  && rimraf packages/commutable/node_modules/immutable",
```
Comment (rgbkrk) [inline package.json]: This will be a stopgap for us until the next lerna release (which will probably be today).

Comment (ivanov) [reply package.json]: yeap, I opened #1436 so we remember to do it in case lerna release gets held up.

**Comment by rgbkrk** [comment]
Whoops, because of me there's a merge conflict with the geojson transform. I'll handle that now.

**Comment by rgbkrk** [comment]
Rebased, brought in the background color -> GeoJSON tile support.

**Comment by rgbkrk** [comment]
Yay, new lerna release fixes it for us! Thanks all.

**Review by peggyrayzis — APPROVED** [review]
Everything looks great to me! I'm excited to start working on exporting packages 🎉

**Pull request merged**